File: internal/audio/omni_gain_test.go
package audio

import "testing"

func TestOmniGain(t *testing.T) {
	gain := NewOmniGain()
	if gain.Name() != "OmniGain" {
		t.Errorf("Expected name OmniGain, got %s", gain.Name())
	}

	gain.SetVolume(0.5)

	buffer := make([]float32, 10)
	for i := range buffer {
		buffer[i] = 1.0
	}

	gain.Process(buffer)

	for i, v := range buffer {
		if v != 0.5 {
			t.Errorf("Expected 0.5 at index %d, got %f", i, v)
		}
	}
}

File: java/src/main/java/com/bqt/audio/OmniGainTest.java
package com.bqt.audio;

public class OmniGainTest {
    public static void main(String[] args) {
        testOmniGain();
        System.out.println("OmniGain tests passed.");
    }

    private static void testOmniGain() {
        OmniGain gain = new OmniGain();
        if (!"OmniGain".equals(gain.getName())) {
            throw new RuntimeException("Expected name OmniGain, got " + gain.getName());
        }

        gain.setVolume(0.5f);

        float[] buffer = new float[10];
        for (int i = 0; i < buffer.length; i++) {
            buffer[i] = 1.0f;
        }

        gain.process(buffer);

        for (int i = 0; i < buffer.length; i++) {
            if (Math.abs(buffer[i] - 0.5f) > 0.000001f) {
                throw new RuntimeException("Expected 0.5 at index " + i + ", got " + buffer[i]);
            }
        }
    }
}
